server/sonar-search/src/main/java/org/sonar/search/EsSettings.java

```diff
@@ -158,21 +183,38 @@ private static void configureIndexDefaults(Settings.Builder builder) {
   }
 
   private void configureCluster(Settings.Builder builder) {
-    int replicationFactor = 0;
+    int replicationFactor = props.valueAsInt(ProcessProperties.SEARCH_REPLICAS, 0);
+    int minimumMasterNodes = props.valueAsInt(ProcessProperties.SEARCH_MINIMUM_MASTER_NODES, 1);
```

Comment:
Same comment, default value should be 1 and not customizable when cluster is disabled.